provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/practitionerrole/PractitionerRoleController.java

```diff
@@ -1,3 +1,150 @@
 
-public class PractitionerRoleController {}
+import gov.va.api.health.providerdirectory.api.resources.PractitionerRole;
+import gov.va.api.health.providerdirectory.service.PpmsPractitionerRole;
+import gov.va.api.health.providerdirectory.service.PpmsProviderSpecialtiesResponse;
+import gov.va.api.health.providerdirectory.service.ProviderContacts;
+import gov.va.api.health.providerdirectory.service.ProviderResponse;
+import gov.va.api.health.providerdirectory.service.client.PpmsClient;
+import gov.va.api.health.providerdirectory.service.controller.Bundler;
+import gov.va.api.health.providerdirectory.service.controller.Bundler.BundleContext;
+import gov.va.api.health.providerdirectory.service.controller.PageLinks.LinkConfig;
+import gov.va.api.health.providerdirectory.service.controller.Parameters;
+import java.util.Collections;
+import java.util.function.Function;
+import javax.validation.constraints.Min;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.MultiValueMap;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Request Mappings for Practitioner Role Profile, see
+ * https://www.fhir.org/guides/argonaut/pd/StructureDefinition-argo-practitionerrole.html for
+ * implementation details.
+ */
+@SuppressWarnings("WeakerAccess")
+@Validated
+@RestController
+@RequestMapping(
+  value = {"/api/PractitionerRole"},
+  produces = {"application/json", "application/json+fhir", "application/fhir+json"}
+)
+@AllArgsConstructor(onConstructor = @__({@Autowired}))
+public class PractitionerRoleController {
+  private PpmsClient ppmsClient;
+
+  private Transformer transformer;
+
+  private Bundler bundler;
+
+  private PractitionerRole.Bundle bundle(
+      MultiValueMap<String, String> parameters, int page, int count) {
+    ProviderResponse providerResponse;
+    if (parameters.get("identifier") != null) {
+      String identifier = parameters.get("identifier").toArray()[0].toString();
+      providerResponse = ppmsClient.providersForId(identifier);
+    } else if (parameters.get("name") != null) {
+      String name = parameters.get("name").toArray()[0].toString();
+      providerResponse = ppmsClient.providersForName(name);
+    } else {
+      String familyAndGiven =
+          parameters.get("family").toArray()[0].toString()
+              + ", "
+              + parameters.get("given").toArray()[0].toString();
+      providerResponse = ppmsClient.providersForName(familyAndGiven);
+    }
+    String providerIdentifier = providerResponse.value().get(0).providerIdentifier().toString();
```

Comment:
If we didn't find a provider, or we did, but the identifier is null for some reason, we should handle those cases and return an empty bundle.